.github/workflows/tests.yaml

```diff
@@ -0,0 +1,33 @@
+name: Run tests
+on: [push, pull_request]
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout pico-sdk/develop
+      uses: actions/checkout@v2
+      with:
+        repository: raspberrypi/pico-sdk
+        ref: develop
+        path: pico-sdk
+
+    - name: Checkout pico-sdk submodules
+      run: cd pico-sdk && git submodule update --init
+
+    - name: Install toolchain dependencies
```

Comment:
I wonder if it's worth moving this to the first build-step, as I guess theoretically we could cache this `apt` step, but we can't really cache the `git` steps? :thinking: